bchain/coins/eth/ethrpc.go

```diff
@@ -56,23 +56,24 @@ type Configuration struct {
 // EthereumRPC is an interface to JSON-RPC eth service.
 type EthereumRPC struct {
 	*bchain.BaseChain
-	Client               bchain.EVMClient
-	RPC                  bchain.EVMRPCClient
-	MainNetChainID       Network
-	Timeout              time.Duration
-	Parser               *EthereumParser
-	PushHandler          func(bchain.NotificationType)
-	OpenRPC              func(string) (bchain.EVMRPCClient, bchain.EVMClient, error)
-	Mempool              *bchain.MempoolEthereumType
-	mempoolInitialized   bool
-	bestHeaderLock       sync.Mutex
-	bestHeader           bchain.EVMHeader
-	bestHeaderTime       time.Time
-	NewBlock             bchain.EVMNewBlockSubscriber
-	newBlockSubscription bchain.EVMClientSubscription
-	NewTx                bchain.EVMNewTxSubscriber
-	newTxSubscription    bchain.EVMClientSubscription
-	ChainConfig          *Configuration
+	Client                  bchain.EVMClient
+	RPC                     bchain.EVMRPCClient
+	MainNetChainID          Network
+	Timeout                 time.Duration
+	Parser                  *EthereumParser
+	PushHandler             func(bchain.NotificationType)
+	OpenRPC                 func(string) (bchain.EVMRPCClient, bchain.EVMClient, error)
+	GetInternalDataForBlock func(blockHash string, blockHeight uint32, transactions []bchain.RpcTransaction) ([]bchain.EthereumInternalData, []bchain.ContractInfo, error)
```

Comment:
support overridable `GetInternalDataForBlock` to support the use of the parity `trace` module for debug. This is only used by gnosis at this time, but I suppose I could also add that logic directly to `ethrpc.go` and leverage a `DebugType` flag if you would like that better.